pkg/audit/p2_balancer.go

```diff
@@ -0,0 +1,180 @@
+package audit
+
+import (
+	"encoding/json"
+
+	"github.com/square/p2/pkg/util"
+
+	pcfields "github.com/square/p2/pkg/pc/fields"
+	rcfields "github.com/square/p2/pkg/rc/fields"
+	"github.com/square/p2/pkg/types"
+)
+
+type P2BalancerErrorMessage string
+type P2SpinJobID string
+
+const (
+	// P2BalancerInProgressEvent denotes the start of a P2 Balancer pod move
+	P2BalancerInProgressEvent EventType = "P2_BALANCER_START"
+
+	// P2BalancerCompletionEvent denotes the successful completion of a
+	// pod move
+	P2BalancerCompletionEvent EventType = "P2_BALANCER_COMPLETION"
+
+	// P2BalancerFailureEvent denotes a pod move failure due
+	// to unrecoverable errors
+	P2BalancerFailureEvent EventType = "P2_BALANCER_FAILURE"
+)
+
+type CommonP2BalancerDetails struct {
+	// P2SpinJobID is a uuid that will be the same for all audit log records
+	// associated with the same pod move. It can be used to match up
+	// "in progress" events with "failure" or "completion" events
+	P2SpinJobID P2SpinJobID `json:"p2_spin_job_id"`
```

Comment:
Is this showing up bleeding too much of our internal pieces to the open source project? What would be an open source user's equivalent? I don't see other mentions of spincycle in the open source project. Do we get around this issue entirely by making it something like `ChangeJobID`?

Reply:
This could be acceptable but keep in mind that spincycle is open sourced but this does expose information.